plugins/modules/ecs_task.py

```diff
@@ -245,6 +264,15 @@ def format_network_configuration(self, network_config):
                 except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
                     self.module.fail_json_aws(e, msg="Couldn't look up security groups")
             result['securityGroups'] = groups
+        if 'assign_public_ip' in network_config:
+            if self.module.botocore_at_least('1.8.4'):
```

Comment:
It's generally better to handle the botocore_at_least pieces as early as possible.  This makes the predicatible failures happen as early as possible.

You can see examples of this down around line 403 (originally line 375), it would be best to handle this down around line 403 rather than here.